components/script/dom/htmlelement.rs

```diff
@@ -151,7 +151,12 @@ impl HTMLElementMethods for HTMLElement {
     // https://html.spec.whatwg.org/multipage/#handler-onload
     fn GetOnload(&self) -> Option<Rc<EventHandlerNonNull>> {
         if self.is_body_or_frameset() {
-            window_from_node(self).GetOnload()
+            let document = document_from_node(self);
+            if document.has_browsing_context() {
+                document.window().GetOnload()
+            } else {
+                None
```

Comment:
The branch we are discussing deals with `body` and `frameset` elements that do not have a browsing context only, right?

My understanding is that `body` and `frameset` elements actually expose event handlers of their Window as their event handlers (first note in [the spec](https://html.spec.whatwg.org/multipage/webappapis.html#event-handler-idl-attributes)).

Is there a case where the note above does not hold?